cpu/sam0_common/periph/gpio.c

```diff
@@ -21,6 +21,18 @@
  * @author      Kaspar Schleiser <kaspar@schleiser.de>
  * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
  *
+ * The GPIO implementation here support the PERIPH_GPIO_FAST_READ pseudomodule
+ * on the cortex-m0 based devices.  This trades an increase in power consumption
+ * for a decrease in GPIO pin read latency. Enable this with:
+ *
+ * ```
+ * USEMODULE += periph_gpio_fast_read
```

Comment:
You want to add `FEATURES_PROVIDED += periph_gpio_fast_read` to `saml1x` and `samd21`.
(Don't forget Kconfig :wink:)

Then we can simply do 

    FEATURES_OPTIONAL += periph_gpio_fast_read

(I'd add that to `tests/periph_gpio` so this code path is tested)

Reply:
Makes sense, let me know if I forgot one of the sam0 variants

Reply:
Nope, I would have forgotten about saml21, but it's a cm0+ too!
samd5x does not have this feature.

Reply:
I've added the cortex-m23 to the docs here